### PR TITLE
added notification on how courses are unlocked for users with no courses

### PR DESCRIPTION
**Pull request by leonardo-fan**
- toast on top right corner for users with no courses saying how courses on the sidebar are unlocked
Course Selector

### CODE REVIEW

File: frontend/src/pages/CourseSelector/CourseSelector.jsx [hunk 1]
```diff
@@ -17,6 +17,25 @@ const CourseSelector = () => {
...
+        message: "How do I unlock courses on the sidebar?",
+        description: "Quick add courses are made visible when you add their prerequisites to your planner and fulfill UOC requirements. Otherwise all courses can be viewed via the search bar.",
+        duration: 30,
+        className: "text helpNotif",
+        placement: "topRight",
+      });
+    };
+
+    // only open for users with no courses
+    if (Object.keys(courses).length === 0) {
```
Comment (Peedee2002) [inline frontend/src/pages/CourseSelector/CourseSelector.jsx:42]: this depends on courses. add that to the dependency list.

Comment (leonardo-fan) [reply frontend/src/pages/CourseSelector/CourseSelector.jsx:42]: ohhh righto sure

File: frontend/src/pages/CourseSelector/CourseSelector.jsx [hunk 2]
```diff
@@ -17,6 +17,25 @@ const CourseSelector = () => {
     programCode, programName, specialisation, minor,
   } = useSelector((state) => state.degree);
 
+  const { courses } = useSelector((state) => state.planner);
+
+  useEffect(() => {
+    const openNotification = () => {
+      notification.info({
+        message: "How do I unlock courses on the sidebar?",
+        description: "Quick add courses are made visible when you add their prerequisites to your planner and fulfill UOC requirements. Otherwise all courses can be viewed via the search bar.",
```
Comment (gtangelo) [inline frontend/src/pages/CourseSelector/CourseSelector.jsx]: lmao i feel like the message notification is a bit long and takes up more room based on the image haha. Perhaps you could shorten it? If not, its okay!

**Review by gtangelo — APPROVED** [review]
lgtm!

File: frontend/src/pages/CourseSelector/CourseSelector.jsx (hunk 2 above)
Comment (leonardo-fan) [reply frontend/src/pages/CourseSelector/CourseSelector.jsx]: ahaha yeah sure I'll try shorten it

Comment (leonardo-fan) [reply frontend/src/pages/CourseSelector/CourseSelector.jsx]: image

Comment (leonardo-fan) [reply frontend/src/pages/CourseSelector/CourseSelector.jsx]: changed it to this - lemme know if it's g :)

Comment (Peedee2002) [reply frontend/src/pages/CourseSelector/CourseSelector.jsx]: hmm maybe say "courses are shown as you meet the requirements to take them. Courses can also be... " - or similar

Comment (leonardo-fan) [reply frontend/src/pages/CourseSelector/CourseSelector.jsx]: Yeahh that's much more succinct, i'll probably use this "Courses are shown as you meet the requirements to take them. Any course can also be selected via the search bar."

**Pull request merged**